src/rust/engine/src/nodes.rs

```diff
@@ -1033,6 +1033,13 @@ impl Node for NodeKey {
   type Error = Failure;
 
   fn run(self, context: Context) -> NodeFuture<NodeResult> {
+    if let Some(path) = self.fs_subject() {
+      let abs_path = context.core.build_root.join(path);
+      context
+        .core
+        .executor
+        .spawn_and_ignore(context.core.watcher.watch(abs_path));
```

Comment:
From a correctness perspective, we should probably actually wait until this operation has completed to execute the rest of the `Node`: otherwise, there is a possibility that we complete the operation in the Node before the watch is in place, and then miss an invalidating change.